### ISSUE TITLE
Slide export doesn't handle `v-if` in `global-top` or `global-bottom` based on conditional using `currentPage`, `currentLayout`

### ISSUE THREAD

**Issue by ani-per** [slidevjs/slidev#1490]
**Describe the bug**

The exporter seems to fail to handle `v-if` conditionals in `global-top` or `global-bottom` that use `currentPage`, `currentLayout`, and other navigation composables.

My global bottom is intended to only be displayed on non-`cover` slides after the first. This functions properly in browser view but fails to display properly on export.

```vue
<script setup lang="ts">
// import { computed } from "vue";
import { useNav, useSlideContext } from "@slidev/client"

// const { $slidev, $route } = useSlideContext()
const { currentPage, currentLayout } = useNav()
</script>

<template>
  <div v-if="currentPage.value > 1 && currentLayout !== 'cover'" class="slide-footer absolute w-full bottom-0 left-0 p-0 text-sm" style="display: grid; grid-template-columns: 33.33% 33.33% 33.34%; color: #ffffff">
    <div class="footer-comp justify-start" style="display: flex; flex-direction: row; gap: 7px; padding-left: 10px">
      <Item text="Ani Perumalla">
        <carbon:user-avatar-filled-alt />
      </Item>
      <Item text="ani-per.gitlab.io">
        <carbon:link />
      </Item>
    </div>
    <div class="footer-comp justify-center" style="display: flex; flex-direction: row; gap: 4px; padding-top: 4px; padding-bottom: 4px;">
      <img src="./img/logos/PSU_Engineering.png" alt="PSU Mark">
      <img src="./img/logos/PSU_Aero.png" alt="PSU Mark">
    </div>
    <div class="footer-comp justify-end" style="display: flex; flex-direction: row; gap: 7px; padding-right: 10px">
      <Item text="PSU UAS Research Lab">
        <carbon:drone />
      </Item>
      <Item text="<a href='https://purl.psu.edu'>purl.psu.edu</a>">
        <carbon:link />
      </Item>
    </div>
  </div>
</template>
```

This produces a proper global bottom in browser view, but no global bottom in the exported pdf:

![Correct browser view](https://github.com/slidevjs/slidev/assets/49991597/6d5f5601-2a84-46dd-896f-d893521fb10b)

![Incorrect exported pdf slide](https://github.com/slidevjs/slidev/assets/49991597/f9d85634-bd4d-449c-9c3b-5844c54a2bad)

When I remove the conditional `v-if="currentPage.value > 1 && currentLayout !== 'cover'"` from the first `div` in the template, the exporter displays the global bottom correctly:

![Correct exported pdf slide](https://github.com/slidevjs/slidev/assets/49991597/a84b3dc6-813c-41bb-9577-e3f5dbd4317e)

**Desktop**

- OS: Windows 11
- Browser: Brave, Firefox, Edge
- Slidev version: 0.48.8

**Comment by KermanX** [comment]
`useNav()` is the global navigation API, which is a singleton.

`useNav().currentPage` is the **page displayed in the main area**, while `$page.value` is the **page this component is rendered on**. (Refer to https://sli.dev/custom/vue-context#page)

If you want to get the current layout, you can use `$nav.currentLayout.value`.

`$page` and `$nav` are available without importing manually. See https://sli.dev/custom/vue-context .

**Comment by ani-per** [comment]
> `useNav()` is the global navigation API, which is a singleton.
> 
> `useNav().currentPage` is the **page displayed in the main area**, while `$page.value` is the **page this component is rendered on**. (Refer to https://sli.dev/custom/vue-context#page)
> 
> If you want to get the current layout, you can use `$nav.currentLayout.value`.
> 
> `$page` and `$nav` are available without importing manually. See https://sli.dev/custom/vue-context .

Thank you for the clarification. However, I'm having a few issues when trying to correct based on this information.

* I can't use `$page` and `$nav` in `global-top/bottom` in the conditional without importing from `useSlideContext()`. I tried without and the conditional failed like before. Is there an exception for root-level files like `global-top/bottom`?
* Even when I ignore this and use `$page` and `$nav` in `global-top/bottom` imported from `useSlideContext()` in the conditional, the global top/bottom still did not show up on the exported slides even though they were rendered properly in browser view.

**Comment by KermanX** [comment]
> I can't use `$page` and `$nav` in `global-top/bottom` in the conditional without importing from `useSlideContext()`. I tried without and the conditional failed like before. Is there an exception for root-level files like `global-top/bottom`?

After some debugging I found this is because there is a "`useSlideContext()`" in the source code (even it's commented out), and then Slidev won't inject `const { $page, ...} = useSlideContext()` automatically.



> * Even when I ignore this and use `$page` and `$nav` in `global-top/bottom` imported from `useSlideContext()` in the conditional, the global top/bottom still did not show up on the exported slides even though they were rendered properly in browser view.

This is a bug. In print mode, the `global-top` is rendered before `injectionCurrentPage` is provided, so the value fallbacks to the global one. I will fix this ASAP.

**Comment by ani-per** [comment]
> > I can't use `$page` and `$nav` in `global-top/bottom` in the conditional without importing from `useSlideContext()`. I tried without and the conditional failed like before. Is there an exception for root-level files like `global-top/bottom`?
> 
> After some debugging I found this is because there is a "`useSlideContext()`" in the source code (even it's commented out), and then Slidev won't inject `const { $page, ...} = useSlideContext()` automatically in this condition.
> 
> > * Even when I ignore this and use `$page` and `$nav` in `global-top/bottom` imported from `useSlideContext()` in the conditional, the global top/bottom still did not show up on the exported slides even though they were rendered properly in browser view.
> 
> This is a bug. In print mode, the `global-top` is rendered before `injectionCurrentPage` is provided, so the value fallbacks to the global one. I will fix this ASAP.

Thank you for the very quick response! I really appreciate your help, and I'm glad to have helped identify the bugs.

**Comment by KermanX** [comment]
Now we have slide layers for this use case. You can simply migrate global layers to slide layers, which is also described in https://sli.dev/features/global-layers.

**Comment by ani-per** [comment]
> Now we have slide layers for this use case. You can simply migrate global layers to slide layers, which is also described in https://sli.dev/features/global-layers.

(A very belated) thanks! Just commenting here for posterity - using `slide-{top,bottom}.vue` instead of `global-{top,bottom}.vue` does indeed fix this issue as of `v51.3.0`, and the header/footer are now visible in the export even without the `--per-slide` flag hack.
